src/logger.ts

```diff
@@ -0,0 +1,97 @@
+import { Logger, LoggerModes } from '@overnightjs/logger'
+import morgan from 'morgan'
+import { config } from 'node-config-ts'
+import * as rfs from 'rotating-file-stream'
+
+enum LogLevel {
+  debug,
+  important,
+  warn
+}
+
+const logLevelMapping = {
+  'debug': LogLevel.debug,
+  'important': LogLevel.important,
+  'warn': LogLevel.warn
+}
+
+const LOG_LEVEL = logLevelMapping[config.logger.level] || LogLevel.important
+const LOG_TO_FILES = config.logger.logToFiles || false
+const LOG_PATH = config.logger.filePath || '/usr/local/var/log/cas'
+const METRICS_LOG_PATH = config.logger.metricsFilePath || LOG_PATH + '/metrics'
+
+/**
+ * Handles logging
+ * TODO: Update doc comments
+ */
+export class CASLogger {
+  public readonly logLevel: LogLevel
+  public readonly expressLoggers: any[]
+  private consoleLogger: Logger
+  private includeStackTrace: boolean
+
+  constructor(logLevel: LogLevel = LogLevel.warn) {
+    this.consoleLogger = new Logger(LoggerModes.Console, '', true)
+    this.logLevel = logLevel
+    this.includeStackTrace = this.logLevel == LogLevel.debug ? true : false
+  }
+
+  // Used for stream interface
+  public write(content: string | object): void {
+    this.info(content)
+  }
+
+  public info(content: string | object): void {
+    if (this.logLevel > LogLevel.debug) return
+    this.consoleLogger.info(content, this.includeStackTrace)
+  }
+
+  public imp(content: string | object): void {
+    if (this.logLevel > LogLevel.important) return
+    this.consoleLogger.imp(content, this.includeStackTrace)
+  }
+
+  public warn(content: string | object): void {
+    this.consoleLogger.warn(content, this.includeStackTrace)
+  }
+
+  public err(content: string | object): void {
+    this.consoleLogger.err(content, this.includeStackTrace)
+  }
+}
+
+export const accessLogStream = rfs.createStream(`${LOG_PATH}/access.log`, {
+  size: "10M", // rotate every 10 MegaBytes written
+  interval: "1d", // rotate daily
+  compress: "gzip" // compress rotated files
+})
+
+export const debugLogStream = rfs.createStream(`${LOG_PATH}/debug.log`, {
+  size: "10M", // rotate every 10 MegaBytes written
+  interval: "1d", // rotate daily
+  compress: "gzip" // compress rotated files
+})
+
+export const errorLogStream = rfs.createStream(`${LOG_PATH}/error.log`, {
+  size: "10M", // rotate every 10 MegaBytes written
+  interval: "1d", // rotate daily
+  compress: "gzip" // compress rotated files
+})
+
+export const metricsLogStream = rfs.createStream(`${METRICS_LOG_PATH}/out.log`, {
```

Comment:
Separating the metrics log stream from the diagnostic log stream is great, and addresses my earlier concerns about mixing the two nicely.  Just to make sure I understand, the only messages going into the metrics log file for now will be those generated by the `@ClassMiddleware` annotation on the controller types, and there will be one log message for each request to one of the API endpoints.  Is that right?

Reply:
No. API req/resp goes into access.log, which the metrics dashboard will read from in addition to the metrics/out.log.

This will be for things other than http api for example low eth.

Reply:
okay, there's there's no actual usages currently in this PR, correct?

Reply:
yeah im fine to pull out of this pr and just have it in the next one with metrics logging